src/org/sosy_lab/cpachecker/cpa/smg/SMGPredicateManager.java

```diff
@@ -141,18 +186,46 @@ private BooleanFormula addPredicateToFormula(BooleanFormula pFormula, SymbolicRe
     return result;
   }
 
-  public BooleanFormula getPredicateFormula(PredRelation pRelation) {
-    return getPredicateFormula(pRelation, true);
+  public BooleanFormula getPathPredicateFormula(UnmodifiableSMGState pState) {
+    SMGPredicateRelation pRelation = pState.getPathPredicateRelation();
+    BooleanFormula predicateFormula = getPredicateFormula(pRelation, true);
+    predicateFormula = addExplicitFromState(predicateFormula, pState);
+    return predicateFormula;
   }
 
-  public BooleanFormula getErrorPredicateFormula(PredRelation pErrorPredicate, PredRelation
-      pPathRelation) {
+  public BooleanFormula getErrorPredicateFormula(
+      SMGPredicateRelation pErrorPredicate, UnmodifiableSMGState pState) {
     BooleanFormula errorFormula = getPredicateFormula(pErrorPredicate, false);
-    BooleanFormula pathFormula = getPredicateFormula(pPathRelation, true);
+    BooleanFormula pathFormula = getPathPredicateFormula(pState);
+    pathFormula = addExplicitFromState(pathFormula, pState);
     return fmgr.makeAnd(pathFormula, errorFormula);
   }
 
-  private BooleanFormula getPredicateFormula(PredRelation pRelation, boolean conjunction) {
+  private BooleanFormula addExplicitFromState(
+      BooleanFormula pFormula, UnmodifiableSMGState pState) {
```

Comment:
The parameter `pFormula` could be avoided.
Simply return the formula from the state and conjunct in calling scope.